src/rows.jl

```diff
@@ -227,6 +230,26 @@ function Base.show(io::IO, row::Row)
     return nothing
 end
 
+"""
+    schema_field_names(::Type{<:Legolas.Schema})
+
+Get a tuple with the names of the fields of this `Legolas.Schema`, including names that
+have been inherited from this `Legolas.Schema`'s parent schema.
+"""
+schema_field_names(s::Schema) = schema_field_names(typeof(s))
+schema_field_names(::Row{S}) where {S} = schema_field_names(S)
+schema_field_names(::Type{<:Row{S}}) where {S} = schema_field_names(S)
+
+"""
+    schema_field_types(::Legolas.Schema{name,version})
+
+Get a tuple with the types of the fields of this `Legolas.Schema`, including types of fields that
+have been inherited from this `Legolas.Schema`'s parent schema.
+"""
+schema_field_types(s::Schema) = schema_field_types(typeof(s))
+schema_field_types(::Row{S}) where {S} = schema_field_types(S)
+schema_field_types(::Type{<:Row{S}}) where {S} = schema_field_types(S)
```

Comment:
Instead of `schema_field_names` / `schema_field_types`, should we define `Tables.Schema(::Legolas.Schema)` or something similar?

https://tables.juliadata.org/stable/#Tables.Schema is the natural object to use for this purpose in Tables.jl ecosystem IIUC

note that I think `Tables.schema(::Vector{<:Legolas.Row})`might already "just work" on `main` (the lowercase `schema` is a generic accessor). Would be worth checking.

Also we should keep in mind that Legolas Schema fields are explicitly unordered - might be best to call that out in the function docs as well